hack/make/test-buildkit

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env bash
+set -e -o pipefail
+
+[ -n "${TESTDEBUG}" ] && set -x
+
+source hack/make/.integration-test-helpers
+
+if [ ! -z "${TEST_SKIP_BUILDKIT}" ]; then
+	echo buildkit tests skipped according to TEST_SKIP_BUILDKIT
+	exit 0
+fi
+
+(
+	env
+	bundle .integration-daemon-start
+
+	export BIN_PATH="${PWD}/bundles/binary-daemon"
+	export PATH="${BIN_PATH}:${PWD}/bundles/binary-daemon:$PATH"
+
+	# Download buildkit
+	# TODO: change this to moby version of the project after
+	# https://github.com/moby/buildkit/pull/1164 is merged.
+	curl -fsSL -o "${PWD}/buildkit.tar.gz" https://github.com/SamWhited/buildkit/archive/${BUILDKIT_COMMIT}.tar.gz
```

Comment:
I think the `./hack/test integration` will build, and run an image, correct?

Not top-priority, but to keep things more separate, wondering if we should use a similar approach as for the `docker-py` tests, where we build the Dockerfile directly from their repository; https://github.com/moby/moby/blob/5a703ccb469c1bd2c75d80ed9891c432078db316/hack/make/test-docker-py#L54

And then run them; https://github.com/moby/moby/blob/5a703ccb469c1bd2c75d80ed9891c432078db316/hack/make/test-docker-py#L62


The problem with that test is still that we build (and run) it docker-in-docker; I think we should run it on the host, and;

1. build the buildkit (or docker-py) image
2. start a "docker-dev" container, which runs the daemon
3. start the buildkit (or docker-py) image, and give it access to the test daemon running in the docker-dev container
4. run the tests


Step 1. could be skipped if BuildKit would push an integration-test image to Docker Hub, so that we only need to pull and run it

Reply:
> I think the ./hack/test integration will build, and run an image, correct?

That's right

> Not top-priority, but to keep things more separate, wondering if we should use a similar approach as for the docker-py tests, where we build the Dockerfile directly from their repository; 

I hadn't seen this, but I like it much better, thanks! I'll think about how this can be restructured.

I'd love to get rid of the docker-in-docker stuff too, one of the difficulties I've had tweaking these tests is that my local system always has some slight difference from CI, so if I want to actually see if things are working I more or less have to use CI. Getting rid of some of the complexity would go a long way towards making it easier to test things locally.

Reply:
> I'd love to get rid of the docker-in-docker stuff too

I don't mind the docker-in-docker for the regular integration tests, but for these, it means we'll be building the same Dockerfile for each run, and no caching whatsoever (because the container in which it's built is removed afterwards)

So, I haven't tried yet to change this for the docker-py test, but it would be something like either a Make target, or in the jenkinsfile, to spin up the two containers; either sharing the `/var/run/docker.sock`, or (perhaps through a compose-file) have them in the same network so that the client can use a tcp connection

Again; just thinking out loud; figuring out a way to keep the separation as much as possible, but still being able to run the tests of "external" project (docker-py, buildkit, perhaps docker-cli) against a locally built daemon